File: api/src/TodoApi.Api/Models/Todo.cs
namespace TodoApi.Api.Models;

public class Todo
{
    public Guid Id { get; set; }
    public string Title { get; set; } = null!;
    public string? Description { get; set; }
    public bool IsCompleted { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime? CompletedAt { get; set; }
    public DateTime? DueDate { get; set; }
    public Priority Priority { get; set; }
}

public enum Priority { Low = 0, Medium = 1, High = 2 }

File: frontend/src/types/todo.ts
export type Priority = 'Low' | 'Medium' | 'High';

export interface Todo {
  id: string;
  title: string;
  description: string | null;
  isCompleted: boolean;
  createdAt: string;
  completedAt: string | null;
  dueDate: string | null;
  priority: Priority;
}

export interface CreateTodoRequest {
  title: string;
  description?: string;
  dueDate?: string;
  priority?: Priority;
}

export interface UpdateTodoRequest {
  title: string;
  description?: string | null;
  dueDate?: string | null;
  priority: Priority;
}

export type SortOption = 'dueDate-asc' | 'dueDate-desc' | 'priority-desc' | 'priority-asc' | 'created-desc' | 'created-asc';
export type StatusFilter = 'all' | 'active' | 'completed';
export type PriorityFilter = 'all' | 'Low' | 'Medium' | 'High';
